app/assets/javascripts/discourse/lib/show-modal.js.es6

```diff
@@ -12,7 +12,7 @@ export default function(name, opts) {
   //
   // Also, very little value in showing main outlet and header on iOS
```

Comment:
This hack was added because of an iOS bug that was resolved in 11.3 update. It fixed an issue with focus input being displaced (https://meta.discourse.org/t/mobile-safari-login-woes/73099) but it causes scroll position being lost when opening a modal on topic page (https://meta.discourse.org/t/losing-scroll-position-in-a-topic-after-flagging/90183).